src/Uvweb/UvBundle/Entity/UniversityRepository.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Uvweb\UvBundle\Entity;
+
+use Doctrine\ORM\EntityRepository;
+
+/**
+ * UniversityRepository
+ *
+ * This class was generated by the Doctrine ORM. Add your own custom
+ * repository methods below.
+ */
+class UniversityRepository extends EntityRepository
+{
+  public function getWithMinimalInfo() {
+    /*
+    Returns an array of the different universities with the associated
+    number of reviews and average grade 
+    */
+    $qb = $this->createQueryBuilder('u');
+    $qb->where('u.approved = 1');
+    $qb->leftJoin('u.classes', 'c')
+      ->leftJoin('c.comments', 'm')
+      ->select('u')
+      ->addSelect('COUNT(DISTINCT m) as num_comments')
+      ->addSelect('COUNT(DISTINCT c) as num_classes')
+      ->addSelect('AVG(m.globalRate) as avg_rate')
+      ->groupBy('u.id');
+    
+    $universities = $qb->getQuery()->getResult();
+    //var_dump($universities);
+
+    $sortedUniversities = array();
+    foreach($universities as $university) {
+      if (is_null($university["num_comments"])) $university["num_comments"] = 0;
```

Comment:
AFAIK all of the conditions here could be replaced by SQL functions.
See https://github.com/uvweb/UVweb/issues/12#issuecomment-266763416